packages/stateManager/index.ts

```diff
@@ -424,12 +436,27 @@ class OrderManager extends PublicObjectManager<Order> {
         status: Status.Pending,
       };
       await this.store.put(id, o);
+      let openOrders: OrdersByStatus = [];
+
+      try {
+        openOrders = (await this.store.get(Status.Pending)) as OrdersByStatus;
+        openOrders.push(id);
+      } catch (error) {
+        const e = error as Error;
+        if (e.notFound) {
+          openOrders.push(id);
+        } else {
+          throw new Error(e.code);
+        }
+      }
+      await this.store.put(Status.Pending, openOrders);
+
```

Comment:
Since you use this pattern several times see if you can generalize it by making it a private function in this class.